internal/pubsub/gcppubsub/gcppubsub.go

```diff
@@ -160,3 +171,15 @@ func (s *subscription) As(i interface{}) bool {
 	*c = s.client
 	return true
 }
+
+// MessageAs implements driver.Subscription.MessageAs.
+func (s *subscription) MessageAs(m *driver.Message, i interface{}) bool {
```

Comment:
I think converting back and forth is ultimately going to be lossy and unsatisfying.

If we don't make `driver.Message` an interface, then at least it should have a field
```
Raw interface{}
```
that drivers can populate with the raw value.

Reply:
That is not sufficient, as one of the benefits of `As` is that the provider can support multiple raw types. It needs to embed a field here that's an `asFunc`; see `blob.Attributes` for an example.